README.md

```diff
@@ -2,3 +2,117 @@
 
 [![Build Status](https://travis-ci.org/tkoolen/SimpleQP.jl.svg?branch=master)](https://travis-ci.org/tkoolen/SimpleQP.jl)
 [![codecov.io](http://codecov.io/github/tkoolen/SimpleQP.jl/coverage.svg?branch=master)](http://codecov.io/github/tkoolen/SimpleQP.jl?branch=master)
+
+SimpleQP makes it easy to set up and efficiently (ideally, with *zero* allocations) solve instances of a **parameterized family** of quadratic programs.
+
+As an example, we'll use the [OSQP](https://github.com/oxfordcontrol/OSQP.jl) solver is used to solve the following problem:
+
+```
+Minimize ||A x - b||^2
+subject to C x = d
+```
+
+with decision variable vector `x`, and where `A`, `b`, `C`, and `d` are parameters with random values, to be re-sampled each time the problem is re-solved.
+
+Here's the basic problem setup:
+```julia
+using OSQP.MathOptInterfaceOSQP
+optimizer = OSQPOptimizer()
+
+using SimpleQP
+model = Model(optimizer)
+n = 8; m = 2
+
+x = [Variable(model) for _ = 1 : n]
+A = Parameter(rand!, zeros(n, n), model)
```

Comment:
I think it might be worth writing out one of these in the `do-block` notation, which makes it a bit clearer what the update function is supposed to do (i.e. that it takes exactly one argument and modifies it in-place).